maro/rl/modeling_v2/critic_model.py

```diff
@@ -194,15 +194,14 @@ def __init__(self, state_dim: int, action_num: int) -> None:
         self._action_num = action_num
 
     def _get_q_critic(self, states: torch.Tensor, actions: torch.Tensor) -> torch.Tensor:
-        q_matrix = self.q_critic_for_all_actions(states)  # [batch_size, action_num]
-        actions = actions.unsqueeze(dim=1)
-        return q_matrix.gather(dim=1, index=actions).reshape(-1)
+        q_values = self.q_critic_for_all_actions(states, actions)  # [batch_size, action_num]
+        return q_values.reshape(-1)
 
     @property
     def action_num(self) -> int:
         return self._action_num
 
-    def q_critic_for_all_actions(self, states: torch.Tensor) -> torch.Tensor:
+    def q_critic_for_all_actions(self, states: torch.Tensor, actions: torch.Tensor) -> torch.Tensor:
```

Comment:
Inconsistency between function name and function content. `q_critic_for_all_actions` means returning Q values for all possible actions according to the given state, so it should not take actions as inputs.

Reply:
Added a new interface `q_critic` and restored this one.